ddtrace/contrib/sanic/patch.py

```diff
@@ -19,17 +19,21 @@
 
 config._add("sanic", dict(_default_service="sanic", distributed_tracing=True))
 
-SANIC_PRE_21 = None
+SANIC_VERSION = (0, 0, 0)
+
+
+def _get_current_span(request):
+    pin = Pin._find(request.ctx)
+    if not pin or not pin.enabled():
+        return None
+
+    return pin.tracer.current_span()
 
 
 def update_span(span, response):
-    if isinstance(response, sanic.response.BaseHTTPResponse):
-        status_code = response.status
-        response_headers = response.headers
-    else:
-        # invalid response causes ServerError exception which must be handled
-        status_code = 500
-        response_headers = None
+    # invalid response causes ServerError exception which must be handled
+    status_code = getattr(response, "status", 500)
```

Comment:
This is effectively the same behavior as before but without the `isinstance` call.

We are just doing duck typing here instead of explicit sub-class checking.

Maybe better to update the above comment to mention this.